### ISSUE TITLE
Additional Dapr metrics properties when subscriber sends `status=DROP` 

### ISSUE THREAD

**Issue by alicejgibbons** [dapr/dapr#7610]
## In what area(s)?

 /area runtime

## What version of Dapr?

1.13.0/1.12.5

## Expected Behavior

If a subscribing app returns `status=drop` but a 200 response then we need an attribute in the Dapr metrics to differentiate when this was dropped by the app or dropped by the Dapr sidecar. If the app drops this message on purpose then it is still a successful request whereas if the Dapr sidecar drops this message it is after many errors on the sidecar. On app side docs: https://docs.dapr.io/reference/api/pubsub_api/#expected-http-response

Eg. for dotnet if the subscriber returns the following, app can drop messages by doing the following.

``` csharp
return Ok(new {status = "DROP"});
```

## Actual Behavior

Looking at the `dapr_component_pubsub_ingress_count` metric this shows up as `process_status="drop"` regardless of if the message is successfully received by the app (200) and then set with `status=drop` or dropped by Dapr.

```
dapr_component_pubsub_ingress_count{app_id="loyalty-service",component="oms.pubsub",namespace="order-management-system",process_status="drop",topic="orders"} 372
```

## Steps to Reproduce the Problem

Add `status=drop` to subscribing apps and port forward Dapr metrics.

## Release Note

RELEASE NOTE: **ADD** Additional metrics data to differentiate DROP status messages.

**Comment by ckcd** [comment]
I am interested in working on this.

/assign

**Comment by ckcd** [comment]
I am interested in working on this.

@yaron2 @alicejgibbons could you please assign this to me? thanks a lot.

**Comment by yaron2** [comment]
> I am interested in working on this.
> 
> @yaron2 @alicejgibbons could you please assign this to me? thanks a lot.

Done, thank you

**Comment by yaron2** [comment]
Closed via #7652
